### PR TITLE
Toulouse/assemblies order

### PR DESCRIPTION
**Pull request by Quentinchampenois**
#### :tophat: What? Why?

As a user, when navigating on the neighborhoods index, I want to see the assemblies displayed in alphanumeric order for specific assemblies

#### :clipboard: Subtasks
- [x] Add new boolean column 'sort_children' on decidim_assemblies
- [x] Get child assemblies sorted by name in show page
- [x] Add checkbox to enable sort_children on admin assembly form
- [x] Add tests

### CODE REVIEW

File: decidim-assemblies/app/views/decidim/assemblies/assemblies/show.html.erb [hunk 1]
```diff
@@ -86,7 +86,11 @@ edit_link(
         <section id="assemblies-grid" class="section row collapse">
           <h4 class="section-heading"><%= t("children", scope: "decidim.assemblies.show") %></h4>
           <div class="row small-up-1 medium-up-2 large-up-2 card-grid">
-            <%= render partial: "decidim/assemblies/assembly", collection: current_participatory_space.children.published %>
+            <% if current_participatory_space.sort_children? %>
+              <%= render partial: "decidim/assemblies/assembly", collection: current_participatory_space.children.published.sort_children_by %>
+            <% else %>
+              <%= render partial: "decidim/assemblies/assembly", collection: current_participatory_space.children.published %>
+          <% end %>
```
Comment (armandfardeau) [inline decidim-assemblies/app/views/decidim/assemblies/assemblies/show.html.erb:93]: Extract : 
`current_participatory_space.sort_children?`
`current_participatory_space.children.published.sort_children_by`
`current_participatory_space.children.published`
to a concern helper

Comment (Quentinchampenois) [reply decidim-assemblies/app/views/decidim/assemblies/assemblies/show.html.erb]: Hi thank you for your suggestion, I agree with you

File: decidim-assemblies/app/helpers/decidim/assemblies/child_assemblies_helper.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Assemblies
+    # Helpers related to the Assemblies children.
+    module ChildAssembliesHelper
+      # Allows to sort the child assemblies by column if needed.
+      # Returns : ActiveRecord::AssociationRelation of Assemblies or empty
+      def child_assemblies_collection(current_participatory_space, sorted_by = nil)
+        current_participatory_space.children.published unless current_participatory_space.sort_children?
```
Comment (armandfardeau) [inline decidim-assemblies/app/helpers/decidim/assemblies/child_assemblies_helper.rb:10]: Shouldn't this be a guard-clause?

```suggestion
        return current_participatory_space.children.published unless current_participatory_space.sort_children?
        
```

File: decidim-assemblies/app/models/decidim/assembly.rb [hunk 3]
```diff
@@ -142,8 +142,8 @@ def closed?
       closing_date < Date.current
     end
 
-    def self.sort_children_by(column = :title)
-      order(column)
+    def self.sort_children_by(ordered_by = :title)
+      ordered_by.nil? ? order(:title) : order(ordered_by)
```
Comment (armandfardeau) [inline decidim-assemblies/app/models/decidim/assembly.rb:146]: I would prefer a guard-clause on this one, what you think?

Comment (Quentinchampenois) [reply decidim-assemblies/app/models/decidim/assembly.rb:146]: Indeed a guard-clause is better

Comment (armandfardeau) [reply decidim-assemblies/app/models/decidim/assembly.rb:146]: Wait for a second... In which case "ordered_by" is nil here? It seems is either :title or anything else, but never nil.

File: decidim-assemblies/app/models/decidim/assembly.rb [hunk 4]
```diff
@@ -142,6 +142,12 @@ def closed?
       closing_date < Date.current
     end
 
+    # returns the children sorted by "ordered_by" params if the assembly has sort_children set to true
+    def sort_children_by(ordered_by = :title)
+      return children.published unless sort_children?
+      children.published.order(ordered_by)
```
Comment (armandfardeau) [inline decidim-assemblies/app/models/decidim/assembly.rb:148]: Please add an empty line after a guard clause

**Pull request merged**